aiida/backends/tests/cmdline/commands/test_computer.py

```diff
@@ -1,13 +1,357 @@
+        result = self.runner.invoke(setup_computer, options)
+        self.assertIsInstance(result.exception, SystemExit)
+        self.assertIn("already exists", result.output)
+
+    def test_noninteractive_disabled(self):
+        """
+        I check what happens if the computer should be disabled.
+
+        I only check the changes, the rest is already checked in ``test_noninteractive()``.
+        """
+        from aiida.orm import Computer
+
+        options_dict = generate_setup_options_dict({'label': 'computer_disabled',
+                                                    ## Pass the '--disabled' option
+                                                    'disabled': None})
+        options_dict.pop('enabled', None) # Make sure --enabled is not there
+        options = generate_setup_options(options_dict)
+
+        result = self.runner.invoke(setup_computer, options)
+
+        self.assertIsNone(result.exception, result.output[-1000:])
+        new_computer = Computer.get(options_dict['label'])
+        self.assertIsInstance(new_computer, Computer)
+        self.assertFalse(new_computer.is_enabled())
+
+    def test_noninteractive_enabled(self):
+        """
+        I check what happens if the computer should be enabled, explicitly setting
+        --enabled.
+
+        I only check the changes, the rest is already checked in ``test_noninteractive()``.
+        """
+        from aiida.orm import Computer
+
+        options_dict = generate_setup_options_dict({'label': 'computer_enabled'})
+        options_dict.pop('disabled', None) # Make sure 'disabled' is not there
+        options_dict['enabled'] = None # Activate --enabled
+        options = generate_setup_options(options_dict)
+
+        result = self.runner.invoke(setup_computer, options)
+
+        self.assertIsNone(result.exception, result.output[-1000:])
+        new_computer = Computer.get(options_dict['label'])
+        self.assertIsInstance(new_computer, Computer)
+        self.assertTrue(new_computer.is_enabled())
+
+    def test_noninteractive_optional_default_mpiprocs(self):
+        """
+        Test 'verdi setup' and check that if is ok not to specify mpiprocs-per-machine
+        """
+        from aiida.orm import Computer
+
+        options_dict = generate_setup_options_dict({'label': 'computer_default_mpiprocs'})
+        options_dict.pop('mpiprocs-per-machine')
+        options = generate_setup_options(options_dict)
+        result = self.runner.invoke(setup_computer, options)
+
+        self.assertIsNone(result.exception, result.output[-1000:])
+
+        new_computer = Computer.get(options_dict['label'])
+        self.assertIsInstance(new_computer, Computer)
+        self.assertIsNone(new_computer.get_default_mpiprocs_per_machine())
+
+    def test_noninteractive_optional_default_mpiprocs_2(self):
+        """
+        Test 'verdi setup' and check that if is the specified value is zero, it means unspecified
```

Comment:
"... check that if the value specified for mpiprocs-per-machine is zero ..."

Reply:
ok